### ISSUE TITLE
[datadogreceiver] unset/unspecified type results in no processed data

### ISSUE THREAD

**Issue by wiardvanrij** [open-telemetry/opentelemetry-collector-contrib#35752]
### Component(s)

receiver/datadog

## Description

It's not required to set `type`. 

However, not setting the type means that data seems to be completely lost. The receiver does say 'OK', as in it received the request, but it doesn't seem to process it futher.

## Steps to Reproduce

_note of not including the arg `type`_

```
curl  -X POST -H "Content-type: application/json" \
    -d '{ "series" :
         [{"metric":"some_metric",
          "points":[['${current_time}', '${metric_val}']],
        "tags":["host:'foo'"]}
    ]
    }' \
    'https://your-endpoint/api/v1/series'
```

## Expected Result

OTeL receiver parses this and then I can forward it to any remote metric endpoint

## Actual Result

No data gets forwarded

## Extra

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/datadogreceiver/internal/translator/series.go#L66

No `NumberDataPointSlice` gets set in the switch. I find it very hard to parse the logic, but I'm assuming because this 'DPS' doesn't get set, further down the line the data isn't getting used at all: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/datadogreceiver/internal/translator/series.go#L90
Which makes it just a null value. 

Also small nit; there are no unit tests for not setting the type.

### extra part 2

I do sort of get the point of that in an ideal case.. the type is set. How else can we infer data. Yet, it doesn't seem to be a required thing on DD itself so I would actually prefer if _somehow_ we can (maybe best effort) make something of it? 

### Collector version

lastest

**Comment by atoulme** [comment]
Please separate your issue into multiple issues as it seems the "Extra" paragraphs report other errors.
